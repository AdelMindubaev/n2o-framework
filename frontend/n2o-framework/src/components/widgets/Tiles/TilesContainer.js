import React from 'react';
import { compose, mapProps } from 'recompose';
import { withWidgetHandlers } from '../AdvancedTable/AdvancedTableContainer';
import { withContainerLiveCycle } from '../Table/TableContainer';

import widgetContainer from '../WidgetContainer';
import Tiles from './Tiles';

function TilesContainer(props) {
  return <Tiles {...props} />;
}

export default compose(
  widgetContainer(
    {
      mapProps: props => ({
        ...props,
      }),
    },
    'TilesWidget'
  ),
  withContainerLiveCycle,
  withWidgetHandlers,
  mapProps(
    ({
      datasource,
      className,
      widgetId,
      tile,
      colsSm,
      colsMd,
      colsLg,
      width,
      height,
    }) => ({
      className: className,
      id: widgetId,
      tile: tile,
      data: datasource,
      colsSm: colsSm,
      colsMd: colsMd,
      colsLg: colsLg,
      tileWidth: width,
      tileHeight: height,
    })
  )
)(TilesContainer);
