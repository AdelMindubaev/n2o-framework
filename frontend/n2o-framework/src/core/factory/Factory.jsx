import React from 'react'
import PropTypes from 'prop-types'
import isEmpty from 'lodash/isEmpty'

import SecurityCheck from '../auth/SecurityCheck'

function Factory(props, context) {
    const { src, level, security, children, ...rest } = props
    let { component = null } = props

    if (!component) {
        component = context.getComponent(src, level)
    }

    if (component) {
        return isEmpty(security) ? (
            React.createElement(component, rest, children)
        ) : (
            <SecurityCheck
                config={security}
                render={({ permissions }) => (permissions
                    ? React.createElement(component, rest, children)
                    : null)}
            />
        )
    }
    console.error(`Фабрике не удалось найти компонент: ${src} в ${level}`)
    return null
}

Factory.propTypes = {
    src: PropTypes.string.isRequired,
    level: PropTypes.string,
    security: PropTypes.object,
}

Factory.contextTypes = {
    getComponent: PropTypes.func,
}

export default Factory
