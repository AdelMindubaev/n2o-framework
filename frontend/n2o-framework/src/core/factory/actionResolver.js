import { NotFoundFactory } from './NotFoundFactory'

export const actionResolver = (src, { actions = {} }) => actions[src] || NotFoundFactory

export default actionResolver
