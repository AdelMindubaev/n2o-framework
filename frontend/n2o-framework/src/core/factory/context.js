import { createContext } from 'react'

export const FactoryContext = createContext({
    resolveProps(props) { return props },
    getComponent() {},
})
