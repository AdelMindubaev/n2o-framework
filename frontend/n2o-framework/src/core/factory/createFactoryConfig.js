import merge from 'deepmerge'

import controls from '../../components/controls'
import widgets from '../../components/widgets'
import regions from '../../components/regions'
import pages from '../../components/pages'
import headers from '../../components/widgets/Table/headers'
import cells from '../../components/widgets/Table/cells'
import fieldsets from '../../components/widgets/Form/fieldsets'
import fields from '../../components/widgets/Form/fields'
import actions from '../../impl/actions'
import buttons from '../../components/buttons'
import snippets from '../../components/snippets'

export const factories = {
    controls,
    widgets,
    regions,
    pages,
    headers,
    cells,
    fieldsets,
    fields,
    actions,
    snippets,
    buttons,
}

export default function createFactoryConfig(customConfig = {}) {
    return merge(factories, customConfig)
}
