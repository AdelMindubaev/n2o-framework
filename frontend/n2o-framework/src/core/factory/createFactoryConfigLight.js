import merge from 'deepmerge';

import InputText from '../../components/controls/InputText/InputText';
import Checkbox from '../../components/controls/Checkbox/CheckboxControl';
import DatePicker from '../../components/controls/DatePicker/DatePicker';
import DateInterval from '../../components/controls/DatePicker/DateInterval';
import PasswordInput from '../../components/controls/PasswordInput/PasswordInput';
import CheckboxGroup from '../../components/controls/CheckboxGroup/CheckboxGroupControl';
import Html from '../../components/controls/Html/Html';
import InputMask from '../../components/controls/InputMask/InputMask';
import InputNumber from '../../components/controls/InputNumber/InputNumber';
import InputSelectContainer from '../../components/controls/InputSelect/InputSelectContainer';
import InputSelectTreeContainer from '../../components/controls/InputSelectTree/InputSelectTreeContainer';
import N2OSelectContainer from '../../components/controls/N2OSelect/N2OSelectContainer';
import RadioGroupControl from '../../components/controls/RadioGroup/RadioGroupControl';
import SelectContainer from '../../components/controls/Select/SelectContainer';
import OutPutText from '../../components/controls/Output/OutputText';
import TextArea from '../../components/controls/TextArea/TextArea';
import InputHidden from '../../components/controls/InputHidden/InputHidden';
import ButtonUploader from '../../components/controls/FileUploader/ButtonUploader';
import DropZone from '../../components/controls/FileUploader/DropZone';
import Switch from '../../components/controls/Switch/Switch';
import InputMoney from '../../components/controls/InputMoney/InputMoney';
import SliderContainer from '../../components/controls/Slider/Slider';
import Rating from '../../components/controls/Rating/Rating';
import Pills from '../../components/controls/Pills/Pills';
import AutoComplete from '../../components/controls/AutoComplete/AutoComplete';

import HtmlWidget from '../../components/widgets/Html/HtmlWidget';
import FormWidget from '../../components/widgets/Form/FormWidget';
import EditForm from '../../components/widgets/Form/FormWidget';
import WireframeWidget from '../../components/widgets/Wireframe/WireframeWidget';
import AdvancedTableWidget from '../../components/widgets/AdvancedTable/AdvancedTableWidget';
import TreeWidget from '../../components/widgets/Tree/TreeWidget';
import ListWidget from '../../components/widgets/List/ListWidget';

import regions from '../../components/regions';
import pages from '../../components/pages';
import headers from '../../components/widgets/Table/headers';
import cells from '../../components/widgets/Table/cells';
import fieldsets from '../../components/widgets/Form/fieldsets';
import fields from '../../components/widgets/Form/fields';
import actions from '../../impl/actions';
import snippets from '../../components/snippets';
import buttons from '../../components/buttons';

export const factoriesLight = {
  controls: {
    InputText: InputText,
    OutputText: OutPutText,
    Checkbox: Checkbox,
    DatePicker: DatePicker,
    DateInterval: DateInterval,
    PasswordInput: PasswordInput,
    CheckboxGroup: CheckboxGroup,
    Html: Html,
    InputMask: InputMask,
    InputNumber: InputNumber,
    InputSelect: InputSelectContainer,
    InputSelectTree: InputSelectTreeContainer,
    N2OSelect: N2OSelectContainer,
    RadioGroup: RadioGroupControl,
    Select: SelectContainer,
    TextArea: TextArea,
    InputHidden: InputHidden,
    ButtonUploader: ButtonUploader,
    DropZone: DropZone,
    Switch: Switch,
    InputMoney: InputMoney,
    Slider: SliderContainer,
    Rating: Rating,
    Pills: Pills,
    AutoComplete: AutoComplete,
  },
  widgets: {
    HtmlWidget,
    FormWidget,
    EditForm,
    WireframeWidget,
    AdvancedTableWidget,
    ListWidget,
    TreeWidget,
  },
  regions,
  pages,
  headers,
  cells,
  fieldsets,
  fields,
  actions,
  snippets,
  buttons,
};

export default function createFactoryConfigLight(customConfig) {
  return merge(factoriesLight, customConfig);
}
