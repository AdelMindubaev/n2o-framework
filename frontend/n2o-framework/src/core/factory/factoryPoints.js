export default [
  'controls',
  'widgets',
  'regions',
  'layouts',
  'headers',
  'cells',
  'fieldsets',
  'fields',
  'actions',
  'snippets',
  'buttons'
];
