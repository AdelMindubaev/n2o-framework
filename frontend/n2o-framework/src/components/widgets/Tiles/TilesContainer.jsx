import React from 'react'
import { compose, mapProps } from 'recompose'

import { withWidgetHandlers } from '../AdvancedTable/AdvancedTableContainer'
import { withContainerLiveCycle } from '../Table/withContainerLiveCycle'

// eslint-disable-next-line import/no-named-as-default
import Tiles from './Tiles'

function TilesContainer(props) {
    return <Tiles {...props} />
}

export default compose(
    withContainerLiveCycle,
    withWidgetHandlers,
    mapProps(
        ({
            models,
            className,
            widgetId,
            tile,
            colsSm,
            colsMd,
            colsLg,
            tileWidth,
            tileHeight,
            setResolve,
            dispatch,
        }) => ({
            className,
            id: widgetId,
            tile,
            data: models.datasource,
            colsSm,
            colsMd,
            colsLg,
            tileWidth,
            tileHeight,
            onResolve: setResolve,
            dispatch,
        }),
    ),
)(TilesContainer)
