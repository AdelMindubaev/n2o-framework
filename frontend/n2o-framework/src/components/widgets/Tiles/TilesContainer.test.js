import React from 'react';
import configureStore from '../../../store';
import { Provider } from 'react-redux';
import TilesContainer from './TilesContainer';
import { WIDGETS } from '../../../core/factory/factoryLevels';
import metadata from './TilesWidget.meta';
import history from '../../../history';

const store = configureStore({}, history, {});

const setup = propsOverride => {
  return mount(
    <Provider store={store}>
      <TilesContainer
        level={WIDGETS}
        {...metadata['Page_Tiles']}
        {...propsOverride}
        id="Page_Tiles"
      />
    </Provider>
  );
};

describe('<TilesContainer />', () => {
  it('отрисовка', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
