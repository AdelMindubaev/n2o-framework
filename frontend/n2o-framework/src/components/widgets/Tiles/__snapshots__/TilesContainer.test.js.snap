// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TilesContainer /> отрисовка 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(withProps(pure(WidgetContainer)))
    colsLg={4}
    colsMd={3}
    colsSm={2}
    id="Page_Tiles"
    level="widgets"
    src="TilesWidget"
    tile={
      Array [
        Object {
          "className": "p-4",
          "component": Object {
            "src": "ImageCell",
          },
          "fieldKey": "url",
          "id": "tile1",
          "style": Object {},
        },
        Object {
          "className": "mt-1 mb-4",
          "component": Object {
            "src": "TextCell",
          },
          "fieldKey": "text",
          "id": "tile2",
          "style": Object {},
        },
        Object {
          "className": "mt-3",
          "component": Object {
            "src": "RatingCell",
          },
          "fieldKey": "rating",
          "id": "tile3",
          "style": Object {},
        },
      ]
    }
  >
    <withProps(pure(WidgetContainer))
      colsLg={4}
      colsMd={3}
      colsSm={2}
      dispatch={[Function]}
      id="Page_Tiles"
      isAnyTableFocused={false}
      level="widgets"
      src="TilesWidget"
      tile={
        Array [
          Object {
            "className": "p-4",
            "component": Object {
              "src": "ImageCell",
            },
            "fieldKey": "url",
            "id": "tile1",
            "style": Object {},
          },
          Object {
            "className": "mt-1 mb-4",
            "component": Object {
              "src": "TextCell",
            },
            "fieldKey": "text",
            "id": "tile2",
            "style": Object {},
          },
          Object {
            "className": "mt-3",
            "component": Object {
              "src": "RatingCell",
            },
            "fieldKey": "rating",
            "id": "tile3",
            "style": Object {},
          },
        ]
      }
      widget={
        Object {
          "containerId": null,
          "count": 0,
          "dataProvider": undefined,
          "error": null,
          "filter": Object {
            "key": null,
            "type": null,
          },
          "isActive": false,
          "isEnabled": true,
          "isFilterVisible": true,
          "isInit": true,
          "isLoading": true,
          "isResolved": false,
          "isVisible": true,
          "modelPrefix": undefined,
          "page": undefined,
          "pageId": undefined,
          "selectedId": null,
          "size": undefined,
          "sorting": Object {},
          "type": "TilesWidget",
          "validation": undefined,
        }
      }
    >
      <pure(WidgetContainer)
        colsLg={4}
        colsMd={3}
        colsSm={2}
        dispatch={[Function]}
        id="Page_Tiles"
        isActive={false}
        isAnyTableFocused={false}
        isEnabled={true}
        isInit={true}
        isLoading={true}
        level="widgets"
        selectedId={null}
        sorting={Object {}}
        src="TilesWidget"
        tile={
          Array [
            Object {
              "className": "p-4",
              "component": Object {
                "src": "ImageCell",
              },
              "fieldKey": "url",
              "id": "tile1",
              "style": Object {},
            },
            Object {
              "className": "mt-1 mb-4",
              "component": Object {
                "src": "TextCell",
              },
              "fieldKey": "text",
              "id": "tile2",
              "style": Object {},
            },
            Object {
              "className": "mt-3",
              "component": Object {
                "src": "RatingCell",
              },
              "fieldKey": "rating",
              "id": "tile3",
              "style": Object {},
            },
          ]
        }
        type="TilesWidget"
        visible={true}
        widget={
          Object {
            "containerId": null,
            "count": 0,
            "dataProvider": undefined,
            "error": null,
            "filter": Object {
              "key": null,
              "type": null,
            },
            "isActive": false,
            "isEnabled": true,
            "isFilterVisible": true,
            "isInit": true,
            "isLoading": true,
            "isResolved": false,
            "isVisible": true,
            "modelPrefix": undefined,
            "page": undefined,
            "pageId": undefined,
            "selectedId": null,
            "size": undefined,
            "sorting": Object {},
            "type": "TilesWidget",
            "validation": undefined,
          }
        }
      >
        <WidgetContainer
          colsLg={4}
          colsMd={3}
          colsSm={2}
          defaultSorting={Object {}}
          dispatch={[Function]}
          fetchOnInit={true}
          id="Page_Tiles"
          isActive={false}
          isAnyTableFocused={false}
          isEnabled={true}
          isInit={true}
          isLoading={true}
          level="widgets"
          placeholder={false}
          resolveModel={Object {}}
          selectedId={null}
          sorting={Object {}}
          src="TilesWidget"
          tile={
            Array [
              Object {
                "className": "p-4",
                "component": Object {
                  "src": "ImageCell",
                },
                "fieldKey": "url",
                "id": "tile1",
                "style": Object {},
              },
              Object {
                "className": "mt-1 mb-4",
                "component": Object {
                  "src": "TextCell",
                },
                "fieldKey": "text",
                "id": "tile2",
                "style": Object {},
              },
              Object {
                "className": "mt-3",
                "component": Object {
                  "src": "RatingCell",
                },
                "fieldKey": "rating",
                "id": "tile3",
                "style": Object {},
              },
            ]
          }
          type="TilesWidget"
          visible={true}
          widget={
            Object {
              "containerId": null,
              "count": 0,
              "dataProvider": undefined,
              "error": null,
              "filter": Object {
                "key": null,
                "type": null,
              },
              "isActive": false,
              "isEnabled": true,
              "isFilterVisible": true,
              "isInit": true,
              "isLoading": true,
              "isResolved": false,
              "isVisible": true,
              "modelPrefix": undefined,
              "page": undefined,
              "pageId": undefined,
              "selectedId": null,
              "size": undefined,
              "sorting": Object {},
              "type": "TilesWidget",
              "validation": undefined,
            }
          }
        >
          <div
            className=""
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <Placeholder
              loading={false}
              once={true}
              type="list"
            >
              <BaseSpinner
                color="primary"
                delay={400}
                loading={true}
                minSpinnerTimeToShow={250}
                text=""
                transparent={false}
                type="cover"
              >
                <div
                  className="n2o-spinner-wrapper"
                >
                  <lifecycle(getContext(withHandlers(mapProps(TilesContainer))))
                    colsLg={4}
                    colsMd={3}
                    colsSm={2}
                    defaultSorting={Object {}}
                    dispatch={[Function]}
                    fetchOnInit={true}
                    id="Page_Tiles"
                    isActive={false}
                    isAnyTableFocused={false}
                    isEnabled={true}
                    isInit={true}
                    isLoading={true}
                    level="widgets"
                    onActionImpl={[Function]}
                    onFetch={[Function]}
                    onFocus={[Function]}
                    onResolve={[Function]}
                    onSetModel={[Function]}
                    onSort={[Function]}
                    placeholder={false}
                    resolveModel={Object {}}
                    selectedId={null}
                    sorting={Object {}}
                    src="TilesWidget"
                    tile={
                      Array [
                        Object {
                          "className": "p-4",
                          "component": Object {
                            "src": "ImageCell",
                          },
                          "fieldKey": "url",
                          "id": "tile1",
                          "style": Object {},
                        },
                        Object {
                          "className": "mt-1 mb-4",
                          "component": Object {
                            "src": "TextCell",
                          },
                          "fieldKey": "text",
                          "id": "tile2",
                          "style": Object {},
                        },
                        Object {
                          "className": "mt-3",
                          "component": Object {
                            "src": "RatingCell",
                          },
                          "fieldKey": "rating",
                          "id": "tile3",
                          "style": Object {},
                        },
                      ]
                    }
                    type="TilesWidget"
                    visible={true}
                    widget={
                      Object {
                        "containerId": null,
                        "count": 0,
                        "dataProvider": undefined,
                        "error": null,
                        "filter": Object {
                          "key": null,
                          "type": null,
                        },
                        "isActive": false,
                        "isEnabled": true,
                        "isFilterVisible": true,
                        "isInit": true,
                        "isLoading": true,
                        "isResolved": false,
                        "isVisible": true,
                        "modelPrefix": undefined,
                        "page": undefined,
                        "pageId": undefined,
                        "selectedId": null,
                        "size": undefined,
                        "sorting": Object {},
                        "type": "TilesWidget",
                        "validation": undefined,
                      }
                    }
                  >
                    <getContext(withHandlers(mapProps(TilesContainer)))
                      colsLg={4}
                      colsMd={3}
                      colsSm={2}
                      defaultSorting={Object {}}
                      dispatch={[Function]}
                      fetchOnInit={true}
                      id="Page_Tiles"
                      isActive={false}
                      isAnyTableFocused={false}
                      isEnabled={true}
                      isInit={true}
                      isLoading={true}
                      level="widgets"
                      onActionImpl={[Function]}
                      onFetch={[Function]}
                      onFocus={[Function]}
                      onResolve={[Function]}
                      onSetModel={[Function]}
                      onSort={[Function]}
                      placeholder={false}
                      resolveModel={Object {}}
                      selectedId={null}
                      sorting={Object {}}
                      src="TilesWidget"
                      tile={
                        Array [
                          Object {
                            "className": "p-4",
                            "component": Object {
                              "src": "ImageCell",
                            },
                            "fieldKey": "url",
                            "id": "tile1",
                            "style": Object {},
                          },
                          Object {
                            "className": "mt-1 mb-4",
                            "component": Object {
                              "src": "TextCell",
                            },
                            "fieldKey": "text",
                            "id": "tile2",
                            "style": Object {},
                          },
                          Object {
                            "className": "mt-3",
                            "component": Object {
                              "src": "RatingCell",
                            },
                            "fieldKey": "rating",
                            "id": "tile3",
                            "style": Object {},
                          },
                        ]
                      }
                      type="TilesWidget"
                      visible={true}
                      widget={
                        Object {
                          "containerId": null,
                          "count": 0,
                          "dataProvider": undefined,
                          "error": null,
                          "filter": Object {
                            "key": null,
                            "type": null,
                          },
                          "isActive": false,
                          "isEnabled": true,
                          "isFilterVisible": true,
                          "isInit": true,
                          "isLoading": true,
                          "isResolved": false,
                          "isVisible": true,
                          "modelPrefix": undefined,
                          "page": undefined,
                          "pageId": undefined,
                          "selectedId": null,
                          "size": undefined,
                          "sorting": Object {},
                          "type": "TilesWidget",
                          "validation": undefined,
                        }
                      }
                    >
                      <withHandlers(mapProps(TilesContainer))
                        colsLg={4}
                        colsMd={3}
                        colsSm={2}
                        defaultSorting={Object {}}
                        dispatch={[Function]}
                        fetchOnInit={true}
                        id="Page_Tiles"
                        isActive={false}
                        isAnyTableFocused={false}
                        isEnabled={true}
                        isInit={true}
                        isLoading={true}
                        level="widgets"
                        onActionImpl={[Function]}
                        onFetch={[Function]}
                        onFocus={[Function]}
                        onResolve={[Function]}
                        onSetModel={[Function]}
                        onSort={[Function]}
                        placeholder={false}
                        resolveModel={Object {}}
                        selectedId={null}
                        sorting={Object {}}
                        src="TilesWidget"
                        store={
                          Object {
                            "dispatch": [Function],
                            "getState": [Function],
                            "replaceReducer": [Function],
                            "subscribe": [Function],
                            Symbol(observable): [Function],
                          }
                        }
                        tile={
                          Array [
                            Object {
                              "className": "p-4",
                              "component": Object {
                                "src": "ImageCell",
                              },
                              "fieldKey": "url",
                              "id": "tile1",
                              "style": Object {},
                            },
                            Object {
                              "className": "mt-1 mb-4",
                              "component": Object {
                                "src": "TextCell",
                              },
                              "fieldKey": "text",
                              "id": "tile2",
                              "style": Object {},
                            },
                            Object {
                              "className": "mt-3",
                              "component": Object {
                                "src": "RatingCell",
                              },
                              "fieldKey": "rating",
                              "id": "tile3",
                              "style": Object {},
                            },
                          ]
                        }
                        type="TilesWidget"
                        visible={true}
                        widget={
                          Object {
                            "containerId": null,
                            "count": 0,
                            "dataProvider": undefined,
                            "error": null,
                            "filter": Object {
                              "key": null,
                              "type": null,
                            },
                            "isActive": false,
                            "isEnabled": true,
                            "isFilterVisible": true,
                            "isInit": true,
                            "isLoading": true,
                            "isResolved": false,
                            "isVisible": true,
                            "modelPrefix": undefined,
                            "page": undefined,
                            "pageId": undefined,
                            "selectedId": null,
                            "size": undefined,
                            "sorting": Object {},
                            "type": "TilesWidget",
                            "validation": undefined,
                          }
                        }
                      >
                        <mapProps(TilesContainer)
                          colsLg={4}
                          colsMd={3}
                          colsSm={2}
                          defaultSorting={Object {}}
                          dispatch={[Function]}
                          fetchOnInit={true}
                          id="Page_Tiles"
                          isActive={false}
                          isAnyTableFocused={false}
                          isEnabled={true}
                          isInit={true}
                          isLoading={true}
                          level="widgets"
                          onActionImpl={[Function]}
                          onFetch={[Function]}
                          onFocus={[Function]}
                          onResolve={[Function]}
                          onRowClickAction={[Function]}
                          onSetModel={[Function]}
                          onSort={[Function]}
                          placeholder={false}
                          resolveModel={Object {}}
                          selectedId={null}
                          sorting={Object {}}
                          src="TilesWidget"
                          store={
                            Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
                          }
                          tile={
                            Array [
                              Object {
                                "className": "p-4",
                                "component": Object {
                                  "src": "ImageCell",
                                },
                                "fieldKey": "url",
                                "id": "tile1",
                                "style": Object {},
                              },
                              Object {
                                "className": "mt-1 mb-4",
                                "component": Object {
                                  "src": "TextCell",
                                },
                                "fieldKey": "text",
                                "id": "tile2",
                                "style": Object {},
                              },
                              Object {
                                "className": "mt-3",
                                "component": Object {
                                  "src": "RatingCell",
                                },
                                "fieldKey": "rating",
                                "id": "tile3",
                                "style": Object {},
                              },
                            ]
                          }
                          type="TilesWidget"
                          visible={true}
                          widget={
                            Object {
                              "containerId": null,
                              "count": 0,
                              "dataProvider": undefined,
                              "error": null,
                              "filter": Object {
                                "key": null,
                                "type": null,
                              },
                              "isActive": false,
                              "isEnabled": true,
                              "isFilterVisible": true,
                              "isInit": true,
                              "isLoading": true,
                              "isResolved": false,
                              "isVisible": true,
                              "modelPrefix": undefined,
                              "page": undefined,
                              "pageId": undefined,
                              "selectedId": null,
                              "size": undefined,
                              "sorting": Object {},
                              "type": "TilesWidget",
                              "validation": undefined,
                            }
                          }
                        >
                          <TilesContainer
                            colsLg={4}
                            colsMd={3}
                            colsSm={2}
                            tile={
                              Array [
                                Object {
                                  "className": "p-4",
                                  "component": Object {
                                    "src": "ImageCell",
                                  },
                                  "fieldKey": "url",
                                  "id": "tile1",
                                  "style": Object {},
                                },
                                Object {
                                  "className": "mt-1 mb-4",
                                  "component": Object {
                                    "src": "TextCell",
                                  },
                                  "fieldKey": "text",
                                  "id": "tile2",
                                  "style": Object {},
                                },
                                Object {
                                  "className": "mt-3",
                                  "component": Object {
                                    "src": "RatingCell",
                                  },
                                  "fieldKey": "rating",
                                  "id": "tile3",
                                  "style": Object {},
                                },
                              ]
                            }
                          >
                            <ForwardRef(withResizeDetector(Tiles))
                              colsLg={4}
                              colsMd={3}
                              colsSm={2}
                              tile={
                                Array [
                                  Object {
                                    "className": "p-4",
                                    "component": Object {
                                      "src": "ImageCell",
                                    },
                                    "fieldKey": "url",
                                    "id": "tile1",
                                    "style": Object {},
                                  },
                                  Object {
                                    "className": "mt-1 mb-4",
                                    "component": Object {
                                      "src": "TextCell",
                                    },
                                    "fieldKey": "text",
                                    "id": "tile2",
                                    "style": Object {},
                                  },
                                  Object {
                                    "className": "mt-3",
                                    "component": Object {
                                      "src": "RatingCell",
                                    },
                                    "fieldKey": "rating",
                                    "id": "tile3",
                                    "style": Object {},
                                  },
                                ]
                              }
                            >
                              <ResizeDetectorHOC
                                colsLg={4}
                                colsMd={3}
                                colsSm={2}
                                forwardedRef={null}
                                tile={
                                  Array [
                                    Object {
                                      "className": "p-4",
                                      "component": Object {
                                        "src": "ImageCell",
                                      },
                                      "fieldKey": "url",
                                      "id": "tile1",
                                      "style": Object {},
                                    },
                                    Object {
                                      "className": "mt-1 mb-4",
                                      "component": Object {
                                        "src": "TextCell",
                                      },
                                      "fieldKey": "text",
                                      "id": "tile2",
                                      "style": Object {},
                                    },
                                    Object {
                                      "className": "mt-3",
                                      "component": Object {
                                        "src": "RatingCell",
                                      },
                                      "fieldKey": "rating",
                                      "id": "tile3",
                                      "style": Object {},
                                    },
                                  ]
                                }
                              >
                                <ResizeDetector
                                  handleHeight={true}
                                  handleWidth={true}
                                  nodeType="div"
                                  onResize={null}
                                  querySelector={null}
                                  refreshRate={1000}
                                  skipOnMount={false}
                                  targetDomEl={null}
                                  targetRef={
                                    Object {
                                      "current": null,
                                    }
                                  }
                                >
                                  <Tiles
                                    colsLg={4}
                                    colsMd={3}
                                    colsSm={2}
                                    targetRef={
                                      Object {
                                        "current": null,
                                      }
                                    }
                                    tile={
                                      Array [
                                        Object {
                                          "className": "p-4",
                                          "component": Object {
                                            "src": "ImageCell",
                                          },
                                          "fieldKey": "url",
                                          "id": "tile1",
                                          "style": Object {},
                                        },
                                        Object {
                                          "className": "mt-1 mb-4",
                                          "component": Object {
                                            "src": "TextCell",
                                          },
                                          "fieldKey": "text",
                                          "id": "tile2",
                                          "style": Object {},
                                        },
                                        Object {
                                          "className": "mt-3",
                                          "component": Object {
                                            "src": "RatingCell",
                                          },
                                          "fieldKey": "rating",
                                          "id": "tile3",
                                          "style": Object {},
                                        },
                                      ]
                                    }
                                  >
                                    <div
                                      className="n2o-tiles__container col-12"
                                    />
                                  </Tiles>
                                </ResizeDetector>
                              </ResizeDetectorHOC>
                            </ForwardRef(withResizeDetector(Tiles))>
                          </TilesContainer>
                        </mapProps(TilesContainer)>
                      </withHandlers(mapProps(TilesContainer))>
                    </getContext(withHandlers(mapProps(TilesContainer)))>
                  </lifecycle(getContext(withHandlers(mapProps(TilesContainer))))>
                </div>
              </BaseSpinner>
            </Placeholder>
          </div>
        </WidgetContainer>
      </pure(WidgetContainer)>
    </withProps(pure(WidgetContainer))>
  </Connect(withProps(pure(WidgetContainer)))>
</Provider>
`;
